examples/vision/mlpmixer_image_classification.py

```diff
@@ -0,0 +1,275 @@
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/05/30
+Last modified: 2021/05/30
+Description: Implementing the MLP-Mixer model for image classification.
+"""
+
+"""
+## Introduction
+
+This example implements the [MLP-Mixer](https://arxiv.org/abs/2105.01601)
+model, by Ilya Tolstikhin et al., for image classification, and demonstrates it
+on the CIFAR-100 dataset. The MLP-Mixer is an architecture based exclusively on
+multi-layer perceptrons (MLPs), and contains two types of MLP layers:
+
+1. One applied independently to image patches, which mixes the per-location features.
+2. The other applied across patches (across channels), which mixes spatial information.
+
+This example requires TensorFlow 2.4 or higher, as well as
+[TensorFlow Addons](https://www.tensorflow.org/addons/overview),
+which can be installed using the following command:
+
+```shell
+pip install -U tensorflow-addons
+```
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow_addons as tfa
+
+"""
+## Prepare the data
+"""
+
+num_classes = 100
+input_shape = (32, 32, 3)
+
+(x_train, y_train), (x_test, y_test) = keras.datasets.cifar100.load_data()
+
+print(f"x_train shape: {x_train.shape} - y_train shape: {y_train.shape}")
+print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+"""
+## Configure the hyperparameters
+"""
+
+learning_rate = 0.005
+weight_decay = 0.0001
+batch_size = 128
+num_epochs = 50
+dropout_rate = 0.2
+image_size = 64  # We'll resize input images to this size.
+patch_size = 8  # Size of the patches to be extracted from the input images.
+num_patches = (image_size // patch_size) ** 2  # Size of the data array.
+hidden_units = 256  # Number of hidden units.
+num_mixers = 4  # Number of mixer blocks.
+
+print(f"Image size: {image_size} X {image_size} = {image_size ** 2}")
+print(f"Patch size: {patch_size} X {patch_size} = {patch_size ** 2} ")
+print(f"Patches per image: {num_patches}")
+print(f"Elements per patch (3 channels): {(patch_size ** 2) * 3}")
+
+"""
+## Use data augmentation
+"""
+
+data_augmentation = keras.Sequential(
+    [
+        layers.experimental.preprocessing.Normalization(),
+        layers.experimental.preprocessing.Resizing(image_size, image_size),
+        layers.experimental.preprocessing.RandomFlip("horizontal"),
+        layers.experimental.preprocessing.RandomZoom(
+            height_factor=0.2, width_factor=0.2
+        ),
+    ],
+    name="data_augmentation",
+)
+# Compute the mean and the variance of the training data for normalization.
+data_augmentation.layers[0].adapt(x_train)
+
+"""
+## Implement the Multi-layer Perceptron (MLP) module
+"""
+
+
+def create_mlp(hidden_units, dropout_rate):
+    mlp = keras.Sequential(
+        [
+            layers.Dense(units=hidden_units, activation=keras.activations.gelu),
+            layers.Dense(units=hidden_units),
```

Comment:
No activation?

Reply:
`activation=keras.activations.gelu` within the Dense layer